lib/Support/CMakeLists.txt

```diff
@@ -2,16 +2,31 @@
 ##
 ##===----------------------------------------------------------------------===//
 
+set(VERSION_CPP "${CMAKE_CURRENT_BINARY_DIR}/Version.cpp")
+set_source_files_properties("${VERSION_CPP}" PROPERTIES GENERATED TRUE)
 
 add_circt_library(CIRCTSupport
   BackedgeBuilder.cpp
   FieldRef.cpp
   LoweringOptions.cpp
   Path.cpp
   APInt.cpp
+  "${VERSION_CPP}"
 
   ADDITIONAL_HEADER_DIRS
 
   LINK_LIBS PUBLIC
   MLIRIR
   )
+
+#-------------------------------------------------------------------------------
+# Generate Version.cpp
+#-------------------------------------------------------------------------------
+find_first_existing_vc_file("${CIRCT_SOURCE_DIR}" CIRCT_GIT_LOGS_HEAD)
+set(GEN_VERSION_SCRIPT "${CIRCT_SOURCE_DIR}/cmake/modules/GenVersionFile.cmake")
+
+# TODO: Change release tag pattern after public release
```

Comment:
I don't know what LLVM does, but it might be good to follow that pattern.

Reply:
The _release tag pattern_ in my comment means the glob pattern in `git describe --match <pattern>`. I use this pattern to distinguish between CIRCT release and PyCDE release. LLVM doesn't use git describe for versions, so they don't have a pattern for us to follow.

I add this comment because I don't think CIRCT's release tag will always have the SiFive prefix. If they change the release tag to v0.1.0, they need to change the `-DRELEASE_PATTERN=sifive*` to `-DRELEASE_PATTERN=v*` in line 32.

Reply:
> do you mean we should release CIRCT like llvmorg-0.1.0?

That would be a good option, or we could use `circtorg-0.1.0`, or `circt-0.1.0`, or even something like `v0.1.0`. I am mainly suggesting we don't use a default for the whole project that is tied to one company. Since this is configurable, SiFive should be able to keep using one tag scheme if that's desired, but I think it would be good to pick a neutral default.

Perhaps this is a topic we should bring up on Discourse for more visibility, or discuss in the ODM on Wednesday.

Reply:
> I think it would be good to pick a neutral default.

I agree. That's why I made this variable configurable.

> Perhaps this is a topic we should bring up on Discourse for more visibility, or discuss in the ODM on Wednesday.

I added a bullet [here](https://docs.google.com/document/d/1fOSRdyZR2w75D87yU2Ma9h2-_lEPL4NxvhJGJd-s5pk/edit#heading=h.252mtx9fm1b). (Is this the ODM you are referring to?)

Reply:
Yep, that's the one, thanks!